### DISCUSSION TITLE
Publishing an Angular library?

### DISCUSSION THREAD

**Discussion by pongells** [yarnpkg/berry#3974] [Q&A]
Hello, 

I have an Angular monorepo managed via lerna and yarn. I am now switching away from lerna, using `yarn workspaces` instead. Everything was going great, until I got to the point of publishing my libraries.

I was expecting `yarn npm publish` to allow publishing from a `dist` folder containing the built sources, but it seems there is no such thing.

Am I missing something? 

Thank you.

**Comment by arcanis** [comment]
It's not clear what's the problem you have. Why isn't running `yarn npm publish` from the workspace you want to publish enough?

**Reply by pongells** [reply]
Because the workspace contains typescript sources which I do not want to publish.

The sources need to be built (via the angular cli), and the result has to then be packaged and published.

**Reply by arcanis** [reply]
You can filter the files that must be part of your published package via the `files` field in your package.json (or alternatively a `.npmignore` file, but `files` is preferred).
